[1.13.x] Refactor run configs based on my FG pr: MinecraftForge/ForgeGradle#546

OhYea777 opened the pull request:
This PR is the result of the combination of two of my other PR's: MinecraftForge/MinecraftForge#5333 and MinecraftForge/MinecraftForge#5471. I have closed those two PR's to centralize them into this.

This updates Forge's `build.gradle` file to better use the run config setup I have created in my FG pr: MinecraftForge/ForgeGradle#546. It also fixes the current issue of using the wrong asset index and now sets it based on the downloaded version meta from FG.

I have also updated parts of the MDK to show, and describe, what can be done with the new run config setup. It also adds a test mod, using the MDK's test source set, to show the ability of multiple mods in a single project. Run configs _should_ also be backwards compatible

LexManos commented on mdk/build.gradle:
```diff
@@ -1,78 +1,166 @@
     repositories {
-        maven { url = 'https://files.minecraftforge.net/maven' }
+        // Try to resolve locally first
+        mavenLocal()
```
For the MDK example, this is bad. As its only for testing things. It'll cause more issues then it's worth for people who don't know what they are doing.

OhYea777 committed:
Refactor run configs based on my FG pr: MinecraftForge/ForgeGradle#546

 > Updated run config formats based on the refactored setup in my FG pr.
 > Removed old 'runclient' and 'runserver' JavaExec tasks, replacing
 them by creating the run configs using the refactored setup.
 > Update to set 'assetIndex' environment variable from FG..
 > Updated MDK for new run config setup (it _should_ work without
 changing anything, however).

OhYea777 commented:
Resolved conflicts and simplified the MDK changes that I had originally made
